src/turtle/env/model/TestNode.d

```diff
@@ -0,0 +1,273 @@
+    import ocean.io.device.File;
+    import ocean.core.Buffer;
+    import ocean.net.server.SelectListener;
+    import ocean.task.Scheduler;
+    import ocean.task.Task;
+    import turtle.env.Shell;
+    import Integer = ocean.text.convert.Integer_tango;
+    import ocean.io.select.client.model.ISelectClient : IAdvancedSelectClient;
+    import ocean.net.server.connection.IConnectionHandlerInfo;
+    import ocean.io.select.protocol.generic.ErrnoIOException;
+    import ocean.util.log.Logger;
+
+    /// Enum defining the possibles states of the fake node service.
+    private enum State
+    {
+        Init,
+        Running,
+        Stopped
+    }
+
+    /// State of the fake node service.
+    private State state;
+
+    /// Used to prevent creating multiple fake nodes of the same type.
+    static bool already_created = false;
+
+    /// Flag indicating that unhandled exceptions from the node must be printed
+    /// in test suite trace.
+    private bool log_errors = true;
+
+    /// Logger shared by all turtle nodes of this type.
+    static private Logger log;
+
+    /***************************************************************************
+
+        Static constructor.
+
+    ***************************************************************************/
+
+    static this ( )
+    {
+        log = Log.lookup("testnode");
+    }
+
+    /***************************************************************************
+
+        Constructor
+
+        Params:
+            node = node addres & port
+            neo_port = port of neo listener (same address as above)
+            conn_setup_params = connection handler constructor arguments
+            options = options for the neo node and connection handlers
+            backlog = (see ISelectListener ctor)
+
+    ***************************************************************************/
+
+    public this ( AddrPort node, ushort neo_port,
+        ConnectionSetupParams conn_setup_params, Options options, int backlog )
+    {
+        verify(!already_created, "Can only have one " ~ idup(this.id) ~
+            " per turtle test app");
+        already_created = true;
+
+        auto addr_buf = new char[AddrPort.AddrBufLength];
+        super(node.asNodeItem(addr_buf), neo_port, conn_setup_params, options,
+            backlog);
+        this.error_callback = &this.onError;
+    }
+
+    /***************************************************************************
+
+        Starts the fake node as part of test suite event loop. It will
+        only terminate when whole test suite process dies.
+
+    ***************************************************************************/
+
+    public void start ( )
+    {
+        verify(this.state == State.Init, "Node has already been started");
+        this.state = State.Running;
+        turtle_env_registry.register(this);
+
+        this.register(theScheduler.epoll);
+        if ( Task.getThis() )
+            theScheduler.processEvents();
```

Comment:
Is there a case where this is not running in task? With the turtle, I don't think so, but I might be mistaken.

Reply:
(note that you require that `theScheduler.epoll` is something - perhaps that's worth verifying as well?)